### DISCUSSION TITLE
command (packages\diagrams\frontend\sirius-components-diagrams) npm run build exited (1)

### DISCUSSION THREAD

**Discussion by Mohammad-Si** [eclipse-sirius/sirius-components#3022] [Q&A] [answered]
Hi guys,

During the building of the frontend section, I encountered an error. Unfortunately, I haven't found a solution yet (I am using Windows; it worked on Linux). Has anyone else faced this error?

Regards
<img width="446" alt="Screenshot 2024-01-26 151612" src="https://github.com/eclipse-sirius/sirius-web/assets/72154633/7bc17b9a-f914-42b8-96de-880be7b85cb2">

**Comment by gcoutable** [comment]
Hello,

Did you try to follow the `Build the frontend & backend components and Sirius Web` section [second step](https://github.com/eclipse-sirius/sirius-web#build-the-frontend--backend-components-and-sirius-web) to set the `script-shell` configuration?

Regards,

**Reply by Mohammad-Si** [reply]
Actually, when I run 'npx turbo run build', I get this error.

**Reply by gcoutable** [reply]
Did you try what the `Warning` is saying?

**Reply by Mohammad-Si** [reply]
Yes, while executing 'npm run build' in the '../sirius-components-diagram' directory, I encountered the following error. I am not proficient in frontend development, so I would appreciate any assistance.
Thanks
<img width="440" alt="sirius-components-diagram" src="https://github.com/eclipse-sirius/sirius-web/assets/72154633/714d6c61-4e40-4f26-81c7-b3d7c04844da">

**Reply by gcoutable** [reply]
Can you give me the node version and npm version you are using (`node --version` and `npm --version`)?
Also, can you give me the whole log in a file please, I will look at it.

**Reply by Mohammad-Si** [reply]
node version is "18.7.0" and npm version is "8.15.0".
[logs.zip](https://github.com/eclipse-sirius/sirius-web/files/14082667/logs.zip)

**Reply by gcoutable** [reply]
The log did not give more information as you already did.
Can you file an issue with the log you gave me and also the result of `npm config list --json`? 
The team will look at it.

**Reply by Mohammad-Si** [reply]
Thank you so much.
Here is the whole required files:
[complete log.zip](https://github.com/eclipse-sirius/sirius-web/files/14083110/complete.log.zip)

**Reply by gcoutable** [reply]
I have created the issue https://github.com/eclipse-sirius/sirius-web/issues/3031

**Reply by gcoutable** [reply, marked as answer]
Since sirius-components 2024.1.2 the package `sirius-components-diagram` content has been replaced by what was in the package `sirius-components-diagram-reactflow`.
You should not have this issue anymore with the latest version.

**Reply by Mohammad-Si** [reply]
Now, That works. Thank you for your support.
